doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -392,8 +465,64 @@ stands for :n:`let @ident := fun {+ @binder} => @term in @term’`.
 
 .. index:: match ... with ...
 
-Definition by case analysis
----------------------------
+Definition by cases: match
+--------------------------
+
+.. insertgram term_match pattern_list_or2
+
+.. productionlist:: coq
+   term_match : match `case_item_list_comma` `return_type_opt` with `or_opt` `eqn_list_or_opt` end
+   case_item_list_comma : `case_item_list_comma` , `case_item`
+                        : `case_item`
+   return_type_opt : `return_type`
+                   : `empty`
+   as_return_type : `as_name_opt` `return_type`
+                  : `empty`
+   return_type : return `term100`
+   or_opt : |
+          : `empty`
```

Comment:
Please move the definition of `or_opt` after `in_opt`. This way, we don't mix nonterminals appearing before and after the `with` keyword.